src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -101,59 +71,87 @@ def generate_requests(
 
         return request_params
 
+    @staticmethod
+    def assign_placeholders(value: str, param_values: Dict[str, Any]) -> str:
+        """
+        Finds all the placeholders (indicated by <>) in the passed in value
+        and replaces them with the actual param values
+        """
+        if value and isinstance(value, str):
+            placeholders = re.findall("<(.+?)>", value)
+            for placeholder in placeholders:
+                value = value.replace(
+                    f"<{placeholder}>", str(param_values[placeholder])
+                )
+        return value
+
+    def map_param_values(
+        self,
+        current_request: SaaSRequest,
+        param_values: Dict[str, Any],
+        update_values: Optional[str],
+    ) -> SaaSRequestParams:
+        """
+        Visits path, headers, query, and body params in the current request and replaces
+        the placeholders with the request param values.
+
+        The update_values are added to the body, if available, and the current_request
+        does not specify a body.
+        """
+
+        path: str = self.assign_placeholders(current_request.path, param_values)
+
+        headers: Dict[str, Any] = {}
+        for header in current_request.headers or []:
```

Comment:
There's a case to say we should update `SaaSRequest` to have sane defaults as opposed to optional values. That way we'd not need to make these `or []` checks for instance. It's not something that ought to happen here. What do you think?